### DISCUSSION TITLE
[DONE ✅]  Custom checkout button text per product

### DISCUSSION THREAD

**Discussion by ghost** [btcpayserver/btcpayserver#2232] [General]
Currently, the text of the button [default `Pay for {0}`] is the same for all products in the app.

However, there can be multiple different types of products where a different sentence is more reasonable / grammatically correct.

It would be nice to customize this button text individually for each product, which can be set when creating a new item / in the editor.

**Comment by bolatovumar** [comment]
@MaxHillebrand which button are you referring to specifically? The one in the Point of Sale app?

![Screen Shot 2021-02-08 at 5 42 13 PM](https://user-images.githubusercontent.com/1934678/107304001-0a16a800-6a35-11eb-8837-08e3b9fca32d.png)

**Reply by ghost** [reply]
yes.

**Reply by bolatovumar** [reply]
I see. Should be easy enough to add this if there is interest.

**Reply by ghost** [reply]
thanks, I would appreciate such a feature!

Simple use case example:
- item that is a "real product" which can being purchased, here `Buy` makes sense.
- then another item that is a donation not in exchange for a good, here `Donate` makes sense.
- and finally one item to gift a product to someone else, here `Sponsor` makes sense.

If you want to have all these different item categories displayed in the same store, having a different pay button text is essential.

**Reply by bolatovumar** [reply]
@pavlenex what do you think about this feature? I looked into this and should be fairly straightforward to implement. Yay or nay?

**Reply by pavlenex** [reply]
@bolatovumar Personally, I don't mind it, if there's community interest.
Maybe we can leave it to be voted in a while, then tackle it? Whichever you prefer.

**Reply by ghost** [reply]
fwiw, I offer a 5k sat bounty for whomever makes the pull request of implementing such custom buttons. Payout is upon PR merge.

just a humble and tiny contribution form my side to prove user demand, please consider in your resource allocation.

**Reply by dennisreimann** [reply]
See #2299

**Comment by ghost** [comment]
I finally tried out this new feature [thanks a bunch again!], and I noticed that when I set a custom Buy Button text, that the price of the item is no longer included.

I'm not sure if this is expected?
Maybe I can set the price with some variable reference? 

![screenshot_20210403_174528](https://user-images.githubusercontent.com/30683012/113483569-64b40980-94a4-11eb-8042-f3073acdbff0.png)

**Reply by dennisreimann** [reply]
This should work using the `Pay {0}`  interpolation – see your initial post in this discussion. :)

**Reply by ghost** [reply]
Yes! that was the `variable reference` I was considering above. 
This works - awesome, thanks for clarifying! 
![screenshot_20210403_182455](https://user-images.githubusercontent.com/30683012/113484700-fa9e6300-94a9-11eb-9a58-708d103f5346.png)
